db/db_impl_open.cc

```diff
@@ -1166,40 +1223,20 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
     return s;
   }
   impl->mutex_.Lock();
-  auto write_hint = impl->CalculateWALWriteHint();
+  // auto write_hint = impl->CalculateWALWriteHint();
   // Handles create_if_missing, error_if_exists
   s = impl->Recover(column_families);
   if (s.ok()) {
+    EnvOptions env_options(db_options);
     uint64_t new_log_number = impl->versions_->NewFileNumber();
-    std::unique_ptr<WritableFile> lfile;
-    EnvOptions soptions(db_options);
-    EnvOptions opt_env_options =
-        impl->immutable_db_options_.env->OptimizeForLogWrite(
-            soptions, BuildDBOptions(impl->immutable_db_options_,
-                                     impl->mutable_db_options_));
-    std::string log_fname =
-        LogFileName(impl->immutable_db_options_.wal_dir, new_log_number);
-    s = NewWritableFile(impl->immutable_db_options_.env, log_fname, &lfile,
-                        opt_env_options);
-    if (s.ok()) {
-      lfile->SetWriteLifeTimeHint(write_hint);
-      lfile->SetPreallocationBlockSize(
-          impl->GetWalPreallocateBlockSize(max_write_buffer_size));
-      {
-        InstrumentedMutexLock wl(&impl->log_write_mutex_);
-        impl->logfile_number_ = new_log_number;
-        const auto& listeners = impl->immutable_db_options_.listeners;
-        std::unique_ptr<WritableFileWriter> file_writer(
-            new WritableFileWriter(std::move(lfile), log_fname, opt_env_options,
-                                   impl->env_, nullptr /* stats */, listeners));
-        impl->logs_.emplace_back(
-            new_log_number,
-            new log::Writer(
-                std::move(file_writer), new_log_number,
-                impl->immutable_db_options_.recycle_log_file_num > 0,
-                impl->immutable_db_options_.manual_wal_flush));
-      }
+    log::Writer* new_log = nullptr;
+    const size_t preallocate_block_size =
+        impl->GetWalPreallocateBlockSize(max_write_buffer_size);
```

Comment:
I'm surprised to see that the other place we are using `mutable_cf_options.write_buffer_size` instead of the max of all CFs. It seems to be a bug to me and we should fix it. The preallocate size should be the same, which is the max of all write buffer size for all CFs.

Reply:
In db_impl_open max_write_buffer_size is already calculated based on all CF's however that's not the case in db_impl_write and that needs to be fixed. Not sure if it's best to tackle it in this task. Maybe we can create another task to fix that bug. Let me know.

Reply:
Let's fix it in this PR. They are supposed to be the same.

Reply:
Hmm, it seems to be not very easy. Let me read code a little bit and see whether there is an easy way.

Reply:
There isn't an easy solution here, so let's leave that two different write_buffer_size is used in two code paths and add a comment.
Try to put GetWalPreallocateBlockSize() call inside the CreateWAL() function, with max_write_buffer_size passed in.

Reply:
Will do.

Reply:
Couldn't do it because GetWalPreallocateBlockSize() has to be called within a mutex.